File: client/src/Types/Category.ts
import ParameterBlock from './ParameterBlock';
import Product from './Product';

type Category = {
  id: string;
  name: string;
  description?: string;
  deliveryPrice?: number;
  commonCategoryId?: string;
  products: Product[];
  parameterBlocks: ParameterBlock[];
};

export default Category;

File: server/Domain/Models/Category.cs
namespace Domain.Models
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;

    public class Category
    {
        public Guid Id { get; set; }

        [Required]
        [MaxLength(30)]
        public string Name { get; set; }

        [Required]
        [MaxLength(200)]
        public string Description { get; set; }

        [Required]
        public decimal DeliveryCost { get; set; }

        public Guid CommonCategoryIdFk { get; set; }

        public HashSet<Product> Products { get; set; }

        public HashSet<CategoryParameterBlock> CategoryParameterBlocks { get; set; }

        public HashSet<ParameterBlock> ParameterBlocks { get; set; }

        public CommonCategory CommonCategory { get; set; }
    }
}
